src/DETHRACE/pc-win95/win95net.c

```diff
@@ -0,0 +1,676 @@
+
+    dr_dprintf("PDNetHostGame()");
+    *pHost_address = &gLocal_addr;
+    return 1;
+}
+
+// IDA: int __usercall PDNetJoinGame@<EAX>(tNet_game_details *pDetails@<EAX>, char *pPlayer_name@<EDX>)
+int PDNetJoinGame(tNet_game_details* pDetails, char* pPlayer_name) {
+    LOG_TRACE("(%p, \"%s\")", pDetails, pPlayer_name);
+
+    dr_dprintf("PDNetJoinGame()");
+    return 0;
+}
+
+// IDA: void __usercall PDNetLeaveGame(tNet_game_details *pDetails@<EAX>)
+void PDNetLeaveGame(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    dr_dprintf("PDNetLeaveGame()");
+}
+
+// IDA: void __usercall PDNetHostFinishGame(tNet_game_details *pDetails@<EAX>)
+void PDNetHostFinishGame(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    dr_dprintf("PDNetHostFinishGame()");
+}
+
+// IDA: tU32 __usercall PDNetExtractGameID@<EAX>(tNet_game_details *pDetails@<EAX>)
+tU32 PDNetExtractGameID(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    return ntohs(pDetails->pd_net_info.addr_in.sin_port); // PDGetTotalTime();
+}
+
+// IDA: tPlayer_ID __usercall PDNetExtractPlayerID@<EAX>(tNet_game_details *pDetails@<EAX>)
+tPlayer_ID PDNetExtractPlayerID(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    return ntohs(gLocal_addr_ipx->sin_port);
+}
+
+// IDA: void __usercall PDNetObtainSystemUserName(char *pName@<EAX>, int pMax_length@<EDX>)
+void PDNetObtainSystemUserName(char* pName, int pMax_length) {
+#ifdef _WIN32
+    int size;
+    char buffer[16];
+    BOOL result;
+    uint32_t size;
+    char buffer[16];
+    int result;
+
+#endif
+
+    dr_dprintf("PDNetObtainSystemUserName()\n");
+    // todo
+#if 0
+    size = COUNT_OF(buffer);
+    result = GetComputerNameA(buffer, &size);
+    if (result && size != 0) {
+        strncpy(pName, buffer, pMax_length - 1);
+        pName[pMax_length - 1] = '\0';
+    }
+    while (1) {
+        pName = strpbrk(pName, "_=(){}[]<>!$%^&*/:@~;'#,?\\|`\"");
+        if (*pName == '\0') {
+            break;
+        }
+        *pName = '-';
+    }
+#else
+    strcpy(pName, "Ron Turn");
+#endif
+}
+
+// IDA: int __usercall PDNetSendMessageToPlayer@<EAX>(tNet_game_details *pDetails@<EAX>, tNet_message *pMessage@<EDX>, tPlayer_ID pPlayer@<EBX>)
+int PDNetSendMessageToPlayer(tNet_game_details* pDetails, tNet_message* pMessage, tPlayer_ID pPlayer) {
+    char str[256];
+    // SOCKADDR_IPX_* remote_addr_ipx;
+    LOG_TRACE("(%p, %p, %d)", pDetails, pMessage, pPlayer);
+    NOT_IMPLEMENTED();
+}
+
+// IDA: int __usercall PDNetSendMessageToAllPlayers@<EAX>(tNet_game_details *pDetails@<EAX>, tNet_message *pMessage@<EDX>)
+int PDNetSendMessageToAllPlayers(tNet_game_details* pDetails, tNet_message* pMessage) {
+    char str[256];
+    int i;
+    LOG_TRACE("(%p, %p)", pDetails, pMessage);
+
+    for (i = 0; i < gNumber_of_net_players; ++i) {
+        if (i == gThis_net_player_index) {
+            continue;
+        }
```

Comment:
I see a function call here, returning a `tNet_game_player_info *` here.

Reply:
You mean a call to a function from inside the for loop? I double checked all the `call` instructions, I don't see any extra ones?

Reply:
You're right.
I mixed this one with `PDNetSendMessageToPlayer`.